osquery/tables/system/darwin/smc_keys.cpp

```diff
@@ -315,5 +316,148 @@ QueryData genSMCKeys(QueryContext &context) {
 
   return results;
 }
+
+QueryData getTemperatures(QueryContext &context) {
+  QueryData results;
+
+  SMCHelper smc;
+  if (!smc.open()) {
+    return {};
+  }
+
+  // Two sources considered:
+  // * http://superuser.com/a/967056
+  // * https://goo.gl/9kv6Ny
+  std::map<std::string, std::string> keyToName = {
+    {"TCXC", "PECI CPU"},
+    {"TCXc", "PECI CPU"},
+    {"TC0P", "CPU 1 Proximity"},
+    {"TC0H", "CPU 1 Heatsink"},
+    {"TC0D", "CPU 1 Package"},
+    {"TC0E", "CPU 1"},
+    {"TC0F", "CPU 1"},
+    {"TC1C", "CPU Core 1"},
+    {"TC2C", "CPU Core 2"},
+    {"TC3C", "CPU Core 3"},
+    {"TC4C", "CPU Core 4"},
+    {"TC5C", "CPU Core 5"},
+    {"TC6C", "CPU Core 6"},
+    {"TC7C", "CPU Core 7"},
+    {"TC8C", "CPU Core 8"},
+    {"TCAH", "CPU 1 Heatsink Alt."},
+    {"TCAD", "CPU 1 Package Alt."},
+    {"TC1P", "CPU 2 Proximity"},
+    {"TC1H", "CPU 2 Heatsink"},
+    {"TC1D", "CPU 2 Package"},
+    {"TC1E", "CPU 2"},
+    {"TC1F", "CPU 2"},
+    {"TCBH", "CPU 2 Heatsink Alt."},
+    {"TCBD", "CPU 2 Package Alt."},
+    {"TCSC", "PECI SA"},
+    {"TCSc", "PECI SA"},
+    {"TCSA", "PECI SA"},
+    {"TCGC", "PECI GPU"},
+    {"TCGc", "PECI GPU"},
+    {"TG0P", "GPU Proximity"},
+    {"TG0D", "GPU Die"},
+    {"TG1D", "GPU Die"},
+    {"TG0H", "GPU Heatsink"},
+    {"TG1H", "GPU Heatsink"},
+    {"Ts0S", "Memory Proximity"},
+    {"TM0P", "Mem Bank A1"},
+    {"TM1P", "Mem Bank A2"},
+    {"TM8P", "Mem Bank B1"},
+    {"TM9P", "Mem Bank B2"},
+    {"TM0S", "Mem Module A1"},
+    {"TM1S", "Mem Module A2"},
+    {"TM8S", "Mem Module B1"},
+    {"TM9S", "Mem Module B2"},
+    {"TN0D", "Northbridge Die"},
+    {"TN0P", "Northbridge Proximity 1"},
+    {"TN1P", "Northbridge Proximity 2"},
+    {"TN0C", "MCH Die"},
+    {"TN0H", "MCH Heatsink"},
+    {"TP0D", "PCH Die"},
+    {"TPCD", "PCH Die"},
+    {"TP0P", "PCH Proximity"},
+    {"TA0P", "Airflow 1"},
+    {"TA1P", "Airflow 2"},
+    {"Th0H", "Heatpipe 1"},
+    {"Th1H", "Heatpipe 2"},
+    {"Th2H", "Heatpipe 3"},
+    {"Tm0P", "Mainboard Proximity"},
+    {"Tp0P", "Powerboard Proximity"},
+    {"Ts0P", "Palm Rest"},
+    {"Tb0P", "BLC Proximity"},
+    {"TL0P", "LCD Proximity"},
+    {"TW0P", "Airport Proximity"},
+    {"TH0P", "HDD Bay 1"},
+    {"TH1P", "HDD Bay 2"},
+    {"TH2P", "HDD Bay 3"},
+    {"TH3P", "HDD Bay 4"},
+    {"TO0P", "Optical Drive"},
+    {"TB0T", "Battery TS_MAX"},
+    {"TB1T", "Battery 1"},
+    {"TB2T", "Battery 2"},
+    {"TB3T", "Battery"},
+    {"Tp0P", "Power Supply 1"},
+    {"Tp0C", "Power Supply 1 Alt."},
```

Comment:
Is there a reason some key names are abbreviated? Maybe there's a max-len somewhere in SMC key land? Mostly curious! :dancer: